src/storage/txn/actions/prewrite.rs

```diff
@@ -52,20 +56,20 @@ pub fn prewrite<S: Snapshot>(
         };
 
     if mutation.should_not_write {
-        return Ok(TimeStamp::zero());
+        return Ok((TimeStamp::zero(), old_value));
     }
 
     if lock_status.fallback_from_async_commit() {
```

Comment:
I think we `let old_value = ` here, return `None` earlier and avoid the mutable variable, which is a little more idiomatic